[kube-prometheus-stack] Complex Config Templating in Alertmanager results into helm warning

### Describe the bug a clear and concise description of what the bug is.

Since #1289, its possible to pass an plain string to `.Values.alertmanager.config`.

If a plain string is defined in helm values, helm starts throwing warnings on helm upgrade
```
coalesce.go:220: warning: cannot overwrite table with non table for opsstack.prom-stack.alertmanager.config (map[global:map[resolve_timeout:5m] inhibit_rules:[map[equal:[namespace alertname] source_matchers:[severity = critical] target_matchers:[severity =~ warning|info]] map[equal:[namespace alertname] source_matchers:[severity = warning] target_matchers:[severity = info]] map[equal:[namespace] source_matchers:[alertname = InfoInhibitor] target_matchers:[severity = info]]] receivers:[map[name:null]] route:map[group_by:[namespace] group_interval:5m group_wait:30s receiver:null repeat_interval:12h routes:[map[matchers:[alertname =~ "InfoInhibitor|Watchdog"] receiver:null]]] templates:[/etc/alertmanager/config/*.tmpl]])
```

### What's your helm version?

version.BuildInfo{Version:"v3.11.0", GitCommit:"472c5736ab01133de504a826bd9ee12cbe4e7904", GitTreeState:"clean", GoVersion:"go1.19.5"}

### What's your kubectl version?

Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-21T14:33:49Z", GoVersion:"go1.19.1", Compiler:"gc", Platform:"darwin/amd64"} Kustomize Version: v4.5.7 Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-12T10:49:09Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

kube-prometheus-stack

### What's the chart version?

43.2.1

### What happened?

Helm throw warnings

```
coalesce.go:220: warning: cannot overwrite table with non table for opsstack.prom-stack.alertmanager.config (map[global:map[resolve_timeout:5m] inhibit_rules:[map[equal:[namespace alertname] source_matchers:[severity = critical] target_matchers:[severity =~ warning|info]] map[equal:[namespace alertname] source_matchers:[severity = warning] target_matchers:[severity = info]] map[equal:[namespace] source_matchers:[alertname = InfoInhibitor] target_matchers:[severity = info]]] receivers:[map[name:null]] route:map[group_by:[namespace] group_interval:5m group_wait:30s receiver:null repeat_interval:12h routes:[map[matchers:[alertname =~ "InfoInhibitor|Watchdog"] receiver:null]]] templates:[/etc/alertmanager/config/*.tmpl]])
```

### What you expected to happen?

No warnings, e.g. using distinct value keys.

### How to reproduce it?

helm upgrade kube-prometheus-stack --set alertmanager.config="receivers: []"

### Enter the changed values of values.yaml?

alertmanager:
  config: |-
    receivers: []

### Enter the command that you execute and failing/misfunctioning.

helm upgrade kube-prometheus-stack --set alertmanager.config="receivers: []"

### Anything else we need to know?

The warnings are excepted from helm side. The only solution would be a distinct value key.

--- a/charts/kube-prometheus-stack/Chart.yaml
+++ b/charts/kube-prometheus-stack/Chart.yaml
@@ -21,7 +21,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 44.4.1
+version: 44.5.0
 appVersion: v0.62.0
 kubeVersion: ">=1.16.0-0"
 home: https://github.com/prometheus-operator/kube-prometheus

--- a/charts/kube-prometheus-stack/templates/alertmanager/secret.yaml
+++ b/charts/kube-prometheus-stack/templates/alertmanager/secret.yaml
@@ -13,14 +13,16 @@ metadata:
 {{ include "kube-prometheus-stack.labels" . | indent 4 }}
 data:
 {{- if .Values.alertmanager.tplConfig }}
-{{- if eq (typeOf .Values.alertmanager.config) "string" }}
+{{- if .Values.alertmanager.stringConfig }}
+  alertmanager.yaml: {{ tpl (.Values.alertmanager.stringConfig) . | b64enc | quote }}
+{{- else if eq (typeOf .Values.alertmanager.config) "string" }}
   alertmanager.yaml: {{ tpl (.Values.alertmanager.config) . | b64enc | quote }}
 {{- else }}
   alertmanager.yaml: {{ tpl (toYaml .Values.alertmanager.config) . | b64enc | quote }}
 {{- end }}
 {{- else }}
   alertmanager.yaml: {{ toYaml .Values.alertmanager.config | b64enc | quote }}
-{{- end}}
+{{- end }}
 {{- range $key, $val := .Values.alertmanager.templateFiles }}
   {{ $key }}: {{ $val | b64enc | quote }}
 {{- end }}

--- a/charts/kube-prometheus-stack/values.yaml
+++ b/charts/kube-prometheus-stack/values.yaml
@@ -212,6 +212,13 @@ alertmanager:
     templates:
     - '/etc/alertmanager/config/*.tmpl'
 
+  ## Alertmanager configuration directives (as string type, preferred over the config hash map)
+  ## stringConfig will be used only, if tplConfig is true
+  ## ref: https://prometheus.io/docs/alerting/configuration/#configuration-file
+  ##      https://prometheus.io/webtools/alerting/routing-tree-editor/
+  ##
+  stringConfig: ""
+
   ## Pass the Alertmanager configuration directives through Helm's templating
   ## engine. If the Alertmanager configuration contains Alertmanager templates,
   ## they'll need to be properly escaped so that they are not interpreted by

